apollo-core/src/main/java/com/ctrip/framework/foundation/internals/NetworkInterfaceManager.java

```diff
@@ -122,4 +125,51 @@ private void load() {
 
     m_local = InetAddress.getLoopbackAddress();
   }
+
+  /**
+   * ref:org.springframework.cloud.commons.util.InetUtils
+   *
+   * @return InetAddress
+   */
+  public InetAddress findFirstNonLoopbackAddress() {
+    InetAddress result = null;
+    try {
+      int lowest = Integer.MAX_VALUE;
+      for (Enumeration<NetworkInterface> nics = NetworkInterface.getNetworkInterfaces();
+          nics.hasMoreElements(); ) {
+        NetworkInterface ifc = nics.nextElement();
+        if (ifc.isUp()) {
+          // testing interface:  + ifc.getDisplayName());
+          if (ifc.getIndex() < lowest || result == null) {
+            lowest = ifc.getIndex();
+          } else {
+            continue;
+          }
+
+          for (Enumeration<InetAddress> addrs = ifc
+              .getInetAddresses(); addrs.hasMoreElements(); ) {
+            InetAddress address = addrs.nextElement();
+            if (address instanceof Inet4Address
+                && !address.isLoopbackAddress()) {
+              //found non-loopback interface: +ifc.getDisplayName()
+              result = address;
```

Comment:
Since the method name is `findFirstNonLoopbackAddress`, shouldn't we `break` here once we found one? Otherwise it would `findLastNonLoopbackAddress`?

Reply:
I guess what `first` mean here is of the lowest index